src/Codegen/Generator.hack

```diff
@@ -229,7 +234,21 @@ final class Generator {
         $hack_class_to_graphql_object = Dict\sort_by_key($hack_class_to_graphql_object);
 
         foreach ($classish_objects as $class) {
-            if (!C\is_empty($class->getAttributes())) {
+            if (Str\ends_with($class->getName(), 'Connection')) {
+                // TODO: Assert that any class which subclasses Connection<T> has a name ending in `Connection`.
+                invariant(
+                    is_connection_type(new \ReflectionClass($class->getName())),
+                    '%s must subclass "Connection"',
+                    $class->getName(),
+                );
+                $objects[] = ObjectBuilder::forConnection($class->getName());
+                $type_param = $class->getASTx()
```

Comment:
This is kind of gnarly and won't work on classes which subclass a non-generic connection. E.g., if `FriendConnection` subclasses `UserConnection`, and `UserConnection` extends `Connection<User>`, then we won't be able to deduce that `FriendConnection` extends `Connection<User>` as well.

Maybe we need to walk the inheritance tree? Ew.

Reply:
I like the idea of just using a type constant; making connection leaf classes generic over multiple types isn't something we need to (or even can, I think) support.